sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCoalesceBatches.scala

```diff
@@ -284,16 +257,17 @@ abstract class AbstractGpuCoalesceIterator(origIter: Iterator[ColumnarBatch],
       var numBytes: Long = 0
       var columnSizes: Array[Long] = schema.fields.indices.map(_ => 0L).toArray
       var stringColumnSizes: Array[Long] = stringFieldIndices.map(_ => 0L)
-      var numBatches = 0
 
       // check if there is a batch "on deck" from a previous call to next()
       if (hasOnDeck) {
         val batch = popOnDeck()
-        addBatch(batch)
-        numBatches += 1
         numRows += batch.numRows()
         columnSizes = getColumnSizes(batch)
         numBytes += columnSizes.sum
+        stringColumnSizes = stringFieldIndices.map(i => getColumnDataSize(batch, i, columnSizes(i)))
```

Comment:
This was a bug in the original code where the `stringColumnSizes` did not take into account anything coming from `onDeck`.  This showed up in a unit test after I combined `RemoveEmptyBatchIterator` so most batches will start out coming from `onDeck`